src/FirebaseAuth.js

```diff
@@ -222,24 +222,23 @@
      */
     _routerMethodOnAuthPromise: function(rejectIfAuthDataIsNull) {
       var ref = this._ref;
-      var deferred = this._q.defer();
 
-      function callback(authData) {
-        if (authData !== null) {
-          deferred.resolve(authData);
-        } else if (rejectIfAuthDataIsNull) {
-          deferred.reject("AUTH_REQUIRED");
-        } else {
-          deferred.resolve(null);
+      return this._utils.promise(function(resolve,reject){
+        function callback(authData) {
+          if (authData !== null) {
+            resolve(authData);
```

Comment:
I have found it to be good practice to add a `return` statement before every call to `resolve()` or `reject()` to make sure no code gets run after them. I've had too many bugs where I didn't do this and they are a huge pain to track down.